test/mpi/dtpools/src/dtpools.c

```diff
@@ -0,0 +1,500 @@
+            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__HINDEXED:
+            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__BLOCK_INDEXED:
+            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__BLOCK_HINDEXED:
+            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__SUBARRAY_C:
+            case DTPI_OBJ_LAYOUT_LARGE_BLK_STRD__SUBARRAY_F:
+                if (factor > (basic_type_count / factor)) {
+                    par.core.type_count = basic_type_count / factor;
+                    par.core.type_blklen = factor;
+                } else {
+                    par.core.type_count = factor;
+                    par.core.type_blklen = basic_type_count / factor;
+                }
+                par.core.type_stride = par.core.type_blklen * 4;
+                break;
+            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__VECTOR:
+            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__HVECTOR:
+            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__INDEXED:
+            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__HINDEXED:
+            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__BLOCK_INDEXED:
+            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__BLOCK_HINDEXED:
+            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__SUBARRAY_C:
+            case DTPI_OBJ_LAYOUT_LARGE_CNT_STRD__SUBARRAY_F:
+                if (factor > (basic_type_count / factor)) {
+                    par.core.type_count = factor;
+                    par.core.type_blklen = basic_type_count / factor;
+                } else {
+                    par.core.type_count = basic_type_count / factor;
+                    par.core.type_blklen = factor;
+                }
+                par.core.type_stride = par.core.type_blklen * 4;
+                break;
+            case DTPI_OBJ_LAYOUT_LARGE_BLK_EXT__VECTOR:
+            case DTPI_OBJ_LAYOUT_LARGE_BLK_EXT__HVECTOR:
+                /* change extent using 'MPI_Type_create_resized' */
+                if (factor > (basic_type_count / factor)) {
+                    par.core.type_count = basic_type_count / factor;
+                    par.core.type_blklen = factor;
+                } else {
+                    par.core.type_count = factor;
+                    par.core.type_blklen = basic_type_count / factor;
+                }
+                par.core.type_stride = par.core.type_blklen + 1;
+                bound = DTPI_BOUND_EXT;
+                break;
+            default:
+                err = DTP_ERR_OTHER;
+                fprintf(stdout, "Type layout %d is not defined\n", obj_idx);
+                fflush(stdout);
+        }
+        err = creators[obj_idx] (bound, &par, dtp);
+    } else {
+        switch (obj_idx) {
+            case DTPI_OBJ_LAYOUT_SIMPLE__STRUCT:
+                /* TODO: add other struct layouts here */
+                break;
+            default:
+                err = DTP_ERR_OTHER;
+                fprintf(stdout, "Type layout %d is not defined\n", obj_idx);
+                fflush(stdout);
+        }
```

Comment:
Should the program just return with error `DTP_ERR_OTHER ` when undefined layout is found ? Now it still calls the creator function.
Same problem in the basic pool (line 312).